src/lib/log-forwarding.js

```diff
@@ -0,0 +1,253 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+const rtLib = require('@adobe/aio-lib-runtime')
+const { writeAio, writeEnv } = require('./import')
+const crypto = require('crypto')
+const fs = require('fs-extra')
+
+const SECRET_FIELD_TYPE = 'password'
+const CHECKSUM_FILE = 'log-forwarding-config.sha256'
+
+class LogForwarding {
+  constructor (aioConfig) {
+    this.aioConfig = aioConfig
+  }
+
+  async init () {
+    rtLib.utils.checkOpenWhiskCredentials({ ow: this.aioConfig.runtime })
+    this.logForwarding = await getRTLogForwarding(this.aioConfig.runtime)
+    return this
+  }
+
+  getLocalConfig () {
+    const config = this.aioConfig.project.workspace.log_forwarding
```

Comment:
can this be null?

Reply:
If somebody writes it as `null` then yes. I'm checking for `undefined` now. But we can add more checks to take into account cases when people change the config manually and put something unexpected.

Reply:
you are checking for undefined, what I mean though is that this code will throw if workspace, or project, or aioConfig is null/undefined.
if we were node@14+ then you could use optional chaining 
`const config = this.aioConfig?.project?.workspace?.log_forwarding`

Reply:
I haven't seen "parent" parts to be null and I checked with other code and it doesn't check these levels for null/undefined. But I can do it.